appscripts/canalytics.gs

```diff
@@ -145,40 +153,212 @@ function scoreBreakdownSummaryHeader(){
+          }
+        if (scoutingItem['Pieces Scored on Mid/High'] == "CONES" ){
+            summaryObjects[scoutingItem['Team Number']]['Pieces_Scored'] = "both"
+          }
+      }
+   })  
+  Object.keys(summaryObjects).forEach( k => {
+    if ( summaryObjects[k]['Matches_Played'] > 0 ) {
+      summaryObjects[k]['Avg_Auton_Score'] = summaryObjects[k]['Avg_Auton_Score']/summaryObjects[k]['Matches_Played']
+      summaryObjects[k]['Avg_Auton_GPs'] = summaryObjects[k]['Avg_Auton_GPs']/summaryObjects[k]['Matches_Played']
+      summaryObjects[k]['Avg_Teleop_Score (no links)'] = summaryObjects[k]['Avg_Teleop_Score (no links)']/summaryObjects[k]['Matches_Played']
+      summaryObjects[k]['Avg_Teleop_GPs'] = summaryObjects[k]['Avg_Teleop_GPs']/summaryObjects[k]['Matches_Played']
+      summaryObjects[k]['Avg_Endgame_Score'] = summaryObjects[k]['Avg_Endgame_Score']/summaryObjects[k]['Matches_Played']
+    }
+  })
+
+  return summaryObjects
+}
+
+function specificScoutingSummaryHeader(){
+  return ["Auton Score", "Teleop Score", "Endgame Score", "Total Score"]
+}
+
+function specificScoutingSummary(specificTeamObjects, specificScoutingObjects){
+  Object.keys(specificTeamObjects).forEach( k => {
+      specificTeamObjects[k]['Auton Score']=0;
+      specificTeamObjects[k]['Teleop Score']=0;
+      specificTeamObjects[k]['Endgame Score']=0;
+      specificTeamObjects[k]['Total Score']=0;
+   })
+  Object.keys(specificScoutingObjects).forEach( k => {
+      specificScoutingItem = specificScoutingObjects[k]
+      var autonScore = 0  
+      var teleopScore = 0
+      var endgameScore = 0
+
+      if ( specificScoutingItem['Mobility Bonus'] == true ){
+        autonScore +=3
+      }
+      if ( specificScoutingItem['Docked/Engaged'] == "DOCK" ){
+        autonScore +=8
+      }
+      if ( specificScoutingItem['Docked/Engaged'] == "ENGAGED" ){
+        autonScore +=12
+      }
+      autonScore += specificScoutingItem['Auton High Pieces Scored']*6
+      autonScore += specificScoutingItem['Auton Mid Pieces Scored']*4
+      autonScore += specificScoutingItem['Auton Low Pieces Scored']*3
+      specificTeamObjects[specificScoutingItem['Team Number']['Match Number']]['Auton Score'] = autonScore
+
+      teleopScore += specificScoutingItem['Teleop High Pieces Scored']*5
+      teleopScore += specificScoutingItem['Teleop Mid Pieces Scored']*3
+      teleopScore += specificScoutingItem['Teleop Low Pieces Scored']*2
+      specificTeamObjects[specificScoutingItem['Team Number']['Match Number']]['Teleop Score'] = teleopScore
+
+      if (specificScoutingItem['Endgame Position'] == "PARK" ){
+        endgameScore +=2
+      }
+      if (specificScoutingItem['Endgame Position'] == "DOCK" ){
+        endgameScore +=6
+      }
+      if (specificScoutingItem['Endgame Position'] == "ENGAGED" ){
+        endgameScore +=10
+      }
+      specificTeamObjects[specificScoutingItem['Team Number']['Match Number']]['Endgame Score'] = endgameScore
```

Comment:
specificScoutingItem['Team Number']['Match Number'] 

doesn't exist.  When you have two indexes, it is an object list of objects.

Do you need the team number?  Should be the same for each match, right?  
Maybe you want to add all of the information you want to specificScoutingItem and then use push?  

above:
specificScoutingItem['Endgame Score']+=10
( and anything else we want to write back to the sheet...)

specificTeamObjects.push(specificScoutingItem)